package/base-files/files/bin/board_detect

```diff
@@ -5,8 +5,8 @@ CFG=$1
 [ -n "$CFG" ] || CFG=/etc/board.json
 
 [ -d "/etc/board.d/" -a ! -s "$CFG" ] && {
-	for a in $(ls /etc/board.d/*); do
-		[ -s $a ] || continue;
+	for a in /etc/board.d/*; do
+		[ -s "$a" ] || continue;
```

Comment:
Even though it is unlikely for files in `/etc/board.d/` to have spaces in their filenames, this is quite appropriate.